### PR TITLE
Get rid of boilerplate code in WEB API's CommandBuilder.

### PR DESCRIPTION
**Pull request by groldan**
WEB API's `CommandBuilder` was cumbersome to maintain with so many conditionals.

This PR makes use of Java 8's method arguments to remove all the boilerplate

### CODE REVIEW

File: src/web/api/src/main/java/org/locationtech/geogig/web/api/CommandBuilder.java [hunk 1]
```diff
@@ -55,471 +88,21 @@
...
-        Integer retval = defaultValue;
-        if (val != null) {
-            try {
-                retval = new Integer(val);
-            } catch (NumberFormatException nfe) {
-                throw new CommandSpecException("Invalid value '" + val + "' specified for option: "
-                        + key);
-            }
+        if (command instanceof AbstractWebAPICommand) {
+            ((AbstractWebAPICommand) command).setTransactionId(options.getFirstValue(
```
Comment (emerkle826) [inline src/web/api/src/main/java/org/locationtech/geogig/web/api/CommandBuilder.java]: @jdgarrett @groldan Does it make more sense to move **setTransactionId** and **getTransactionId** to the WebAPICommand interface, so we can avoid calling **instance of** here?

**Comment by emerkle826** [comment]
@groldan @jdgarrett other that the philosophy about *instance of*, this PR looks good to me. :+1:

File: src/web/api/src/main/java/org/locationtech/geogig/web/api/CommandBuilder.java (hunk 1 above)
Comment (groldan) [inline src/web/api/src/main/java/org/locationtech/geogig/web/api/CommandBuilder.java]: Removed need to instanceof check at 5c0c31365a8a2c4a97f27352675751cc17130bcb

**Comment by emerkle826** [comment]
:+1:

**Pull request merged**